core/base/result.go

```diff
@@ -58,13 +58,13 @@ func (s TokenResultStatus) String() string {
 type TokenResult struct {
 	status TokenResultStatus
 
-	blockErr *BlockError
-	waitMs   uint64
+	blockErr    *BlockError
+	nanosToWait uint64
```

Comment:
Maybe use time.Duration is better?  So, we need not to honor the time precision

Reply:
Good idea. Updated.